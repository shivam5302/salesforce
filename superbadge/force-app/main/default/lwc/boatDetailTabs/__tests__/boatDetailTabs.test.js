import { createElement } from 'lwc';
import BoatDetailTabs from 'c/boatDetailTabs';
import { getRecord } from 'lightning/uiRecordApi'
import { subscribe } from 'lightning/messageService';


jest.mock('c/fiveStarRating')

jest.mock('lightning/messageService', () => {
	return {
		subscribe: jest.fn(),
		APPLICATION_SCOPE: { APPLICATION_SCOPE: true },
		MessageContext: jest.fn()
	};
});

const mockCustomEvent = CustomEvent;
jest.mock(
	'lightning/navigation',
	() => {
	  const result = {};
  
	  const Navigate = Symbol("Navigate");
  
	  const NavigationMixin = (Base) => {
		return class extends Base {
		  [Navigate](navigateArg){
			this.dispatchEvent(new mockCustomEvent('navigate', {detail:navigateArg}));
		  }
		};
	  };
	  NavigationMixin.Navigate = Navigate;
  
	  result.NavigationMixin = NavigationMixin;
  
	  return result;
	},
	{ virtual: true }
  );

jest.mock('lightning/uiRecordApi',
	() => {
		const { createApexTestWireAdapter } = require('@salesforce/sfdx-lwc-jest');
		return {
			getRecord: createApexTestWireAdapter(jest.fn()),
			getFieldValue: (boat,field) => {
				
				return boat?.fields.Name.value;
			}
		};
	},
	{ virtual: true }
)

const flushPromises = async () => new Promise((resolve) => setTimeout(resolve, 0));

describe('c-boat-detail-tabs', () => {
	let element;

	beforeEach(async () => {
		element = createElement('c-boat-detail-tabs', {
			is: BoatDetailTabs
		});
		document.body.appendChild(element);

		await flushPromises()
		const mockMessage = { recordId: 'abc123' };
		const messageCallback = subscribe.mock.calls[0][2];
		messageCallback(mockMessage);
	});

	afterEach(() => {
		// The jsdom instance is shared across test cases in a single file so reset the DOM
		while (document.body.firstChild) {
			document.body.removeChild(document.body.firstChild);
		}
		subscribe.mockClear();
	});

	it('should set boat data when getRecord returns data', async () => {
		//Given
		const mockData = {
			fields: {
				Name: { value: 'Test Boat' }
			}
		};

		//When
		getRecord.emit(mockData);
		//Then
		await Promise.resolve();
		// expect(card.title).toBe('Test Boat');
		expect(element).toMatchSnapshot();
	});

	it('should navigate to the boat detail standard page', async () => {
		//Given
		await flushPromises();
		const lighntingButton = element.shadowRoot.querySelector('lightning-button');
		const event = new CustomEvent('click',{detail:''});
		debugger;

		const navigateHandler = jest.fn();
		element.addEventListener('navigate', navigateHandler);
		
		//When
		lighntingButton.dispatchEvent(event);

		//Then
		expect(navigateHandler.mock.calls[0][0].detail).toEqual({
			type: 'standard__recordPage',
			attributes: { objectApiName: 'Boat__c', recordId: "abc123", actionName: 'view' }
		  })

	});
});