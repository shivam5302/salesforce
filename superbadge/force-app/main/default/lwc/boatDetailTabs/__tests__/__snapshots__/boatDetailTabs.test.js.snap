// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`c-boat-detail-tabs should set boat data when getRecord returns data 1`] = `
<c-boat-detail-tabs
  __lwc_scope_token__=""
>
  #shadow-root(open)
    <!---->
    <lightning-tabset
      __lwc_scope_token__=""
    >
      #shadow-root(open)
        <slot />
      <lightning-tab
        __lwc_scope_token__=""
      >
        #shadow-root(open)
          <slot />
        <lightning-card
          __lwc_scope_token__=""
        >
          #shadow-root(open)
            <slot />
            <slot
              name="actions"
            />
            <slot
              name="footer"
            />
            <slot
              name="title"
            />
          <lightning-record-view-form
            __lwc_scope_token__=""
          >
            #shadow-root(open)
              <slot />
            <div
              __lwc_scope_token__=""
              class="inline-field"
            >
              <lightning-output-field
                __lwc_scope_token__=""
              >
                #shadow-root(open)
                  <slot />
              </lightning-output-field>
            </div>
            <div
              __lwc_scope_token__=""
              class="inline-field"
            >
              <lightning-output-field
                __lwc_scope_token__=""
              >
                #shadow-root(open)
                  <slot />
              </lightning-output-field>
            </div>
            <div
              __lwc_scope_token__=""
              class="inline-field"
            >
              <lightning-output-field
                __lwc_scope_token__=""
              >
                #shadow-root(open)
                  <slot />
              </lightning-output-field>
            </div>
            <div
              __lwc_scope_token__=""
              class="inline-field"
            >
              <lightning-output-field
                __lwc_scope_token__=""
              >
                #shadow-root(open)
                  <slot />
              </lightning-output-field>
            </div>
          </lightning-record-view-form>
          <lightning-button
            __lwc_scope_token__=""
            slot="actions"
          >
            #shadow-root(open)
          </lightning-button>
        </lightning-card>
      </lightning-tab>
      <lightning-tab
        __lwc_scope_token__=""
      >
        #shadow-root(open)
          <slot />
        <c-boat-review
          __lwc_scope_token__=""
        >
          #shadow-root(open)
            <!---->
            <div
              __lwc_scope_token__=""
            >
              No reviews available
            </div>
            <!---->
        </c-boat-review>
      </lightning-tab>
      <lightning-tab
        __lwc_scope_token__=""
      >
        #shadow-root(open)
          <slot />
        <c-boat-add-review-form
          __lwc_scope_token__=""
        >
          #shadow-root(open)
            <lightning-card>
              #shadow-root(open)
                <slot />
                <slot
                  name="actions"
                />
                <slot
                  name="footer"
                />
                <slot
                  name="title"
                />
              <lightning-record-edit-form>
                #shadow-root(open)
                  <slot />
                <lightning-input-field>
                  #shadow-root(open)
                </lightning-input-field>
                <lightning-layout>
                  #shadow-root(open)
                    <slot />
                  <lightning-layout-item>
                    #shadow-root(open)
                      <slot />
                    <label
                      class="slds-form-element__label"
                      for="record-name-13"
                    >
                      Review Subject
                    </label>
                    <div
                      class="slds-form-element"
                    >
                      <lightning-input-field>
                        #shadow-root(open)
                      </lightning-input-field>
                    </div>
                  </lightning-layout-item>
                  <lightning-layout-item>
                    #shadow-root(open)
                      <slot />
                    <div
                      class="slds-form-element"
                    >
                      <label
                        class="slds-form-element__label"
                        for="record-name-13"
                      >
                        Rating
                      </label>
                      <div
                        class="slds-form-element__control"
                      >
                        <c-five-star-rating>
                          #shadow-root(open)
                        </c-five-star-rating>
                      </div>
                    </div>
                  </lightning-layout-item>
                  <lightning-layout-item>
                    #shadow-root(open)
                      <slot />
                    <lightning-textarea>
                      #shadow-root(open)
                    </lightning-textarea>
                  </lightning-layout-item>
                </lightning-layout>
                <div
                  class="slds-align_absolute-center"
                  style="margin-top: 5px;"
                >
                  <lightning-button>
                    #shadow-root(open)
                  </lightning-button>
                </div>
              </lightning-record-edit-form>
            </lightning-card>
        </c-boat-add-review-form>
      </lightning-tab>
    </lightning-tabset>
    <!---->
</c-boat-detail-tabs>
`;
